chatkit-core/src/test/kotlin/com/pusher/chatkit/users/UserSubscriptionDispatcherTest.kt

```diff
@@ -0,0 +1,193 @@
+                        room = joinedRoomApiTypeMapper.toRoomInternalType(event.room)
+                )) }
+            }
+        }
+    }
+
+    describe("given a user subscription dispatcher with previous state") {
+
+        val state = mockk<GetState<State>>(relaxed = true)
+        val dispatcher = mockk<Dispatcher>(relaxed = true)
+        val joinedRoomApiTypeMapper = JoinedRoomApiTypeMapper()
+        val userSubscriptionDispatcher = UserSubscriptionDispatcher(
+                stateGetter = state,
+                joinedRoomApiTypeMapper = joinedRoomApiTypeMapper,
+                dispatcher = dispatcher
+        )
+
+        every { state().joinedRoomsState } returns JoinedRoomsState(
+                rooms = mapOf(roomTwoId to roomTwo),
+                unreadCounts = mapOf(roomTwoId to 2))
+
+        describe("when I receive a second InitialState event with a different room") {
+
+            val event = UserSubscriptionEvent.InitialState(
+                    currentUser = simpleUser,
+                    rooms = listOf(simpleJoinedRoomApiType),
+                    readStates = listOf(RoomReadStateApiType("id1", 1, null)),
+                    memberships = listOf(RoomMembershipApiType("id1", listOf()))
+            )
+            userSubscriptionDispatcher.onEvent(event)
+
+            it("then the dispatcher will send a JoinRoom action") {
+                verify(exactly = 1) { dispatcher(JoinedRoom(
+                        room = joinedRoomApiTypeMapper.toRoomInternalType(event.rooms[0]),
+                        unreadCount = 1
+                )) }
+            }
+
+            it("then the dispatcher will send a LeftRoom action") {
+                verify(exactly = 1) { dispatcher(LeftRoom(roomId = "id2")) }
+            }
+        }
+
+        describe("when I receive a second InitialState event with an updated room") {
+            val event = UserSubscriptionEvent.InitialState(
+                    currentUser = simpleUser,
+                    rooms = listOf(JoinedRoomApiType(
+                            id = "id2",
+                            name = "room2-updated",
+                            createdById = "person1",
+                            pushNotificationTitleOverride = "notification override",
+                            private = false,
+                            customData = mapOf("highlight" to "blue"),
+                            lastMessageAt = "2020-02-27T17:12:10Z",
+                            updatedAt = "2020-02-27T17:12:20Z",
+                            createdAt = "2020-02-27T17:12:30Z",
+                            deletedAt = null
+                    )),
+                    readStates = listOf(RoomReadStateApiType("id1", 1, null)),
+                    memberships = listOf(RoomMembershipApiType("id1", listOf()))
+            )
+            userSubscriptionDispatcher.onEvent(event)
+
+            it("then the dispatcher will send an UpdatedRoom action") {
```

Comment:
```suggestion
            it("then an UpdatedRoom action is dispatched") {
```

Reply:
👍 have updated to match for the others too